installer/common.sh

```diff
@@ -0,0 +1,238 @@
+    local prompt
+    prompt=$(sudo -nv 2>&1)
+    if [ $? -eq 0 ]; then
+        : # Has sudo permissions and password entered recently
+    elif echo $prompt | grep -q '^sudo:'; then
+        : # Has sudo permissions but needs password
+    else
+        # No sudo permissions whatsoever
+        echo "No"
+        echo
+        echo "Exiting installer. Run again with a user that has sudo permissions"
+
+        exit 1
+    fi
+
+    echo "Yes"
+    echo
+}
+
+sudo_active() {
+    sudo -nv 2>&1
+}
+
+ask_sudo_pw() {
+    # Skipping because we are root
+    [ "$EUID" -eq 0 ] && return 0
+
+    echo "Everything ready."
+    echo "Please enter your sudo password to proceed..."
+    sudo -v
+}
+
+ask_for_value() {
+    local variable_name="$2"
+    local default_value="${!variable_name}"
+    local question="$1 [$default_value]: "
+    local answer
+
+    read -p "$question" answer
+    echo
+
+    declare -g "$variable_name=${answer:-$default_value}"
+}
+
+ask_yes_no() {
+    local variable_name="$2"
+    local choice_indicator="$(([ -n "${!variable_name}" ] && "${!variable_name}") && echo "Y/n" || echo "y/N")"
+    local question="$1 [$choice_indicator]: "
+    local answer
+
+    read -p "$question" answer
+    while [ -n "$answer" ] && [ "$answer" != "Y" ] && [ "$answer" != "y" ] && [ "$answer" != "N" ] && [ "$answer" != "n" ]; do
+        read -p "Please answer with \"y\" or \"n\" [$choice_indicator]: " answer
+    done
+    echo
+
+    if [ "$answer" == "Y" ] || [ "$answer" == "y" ]; then
+         declare -g "$variable_name=true"
+    elif [ "$answer" == "N" ] || [ "$answer" == "n" ] || [ -z "${!variable_name}" ]; then
+         declare -g "$variable_name=false"
+    fi
+}
+
+command_exists() {
+    local program="$1"
+
+    sudo which "$program" > /dev/null
+}
+
+get_nginx_ip() {
+    if ! command_exists netstat; then
+        echo "autodetect"
+        return 0
+    fi
+
+    local ip_version="$1"
+
+    if [ $# -eq 1 ]; then
+        local ip
+
+        "$USE_HTTPS" && ip="$(get_nginx_ip "$ip_version" 443)"
+        [ -z "$ip" ] && ip="$(get_nginx_ip "$ip_version" 80)"
+        [ -z "$ip" ] && ip="autodetect"
+
+        echo "$ip"
+
+        return 0
+    elif [ $# -ne 2 ]; then
+        return 1
+    fi
+
+    local port="$2"
+
+    # Try detecting the IP nginx listens to (and filter out localhosts)
+    local socket="$(
+        exec $(sudo_active && echo sudo) netstat -tplnW -"$ip_version" 2> /dev/null |
+        grep nginx |
+        grep "$port" |
+        tr -s ' ' |
+        cut -d' ' -f4 |
+        grep -v " 127." |
+        grep -v " ::1:" |
+        head -n1
+    )"
+
+    echo "${socket%:*}"
+}
+
+check_package() {
+    local program="$1"
+    local package="$2"
+
+    if ! command_exists "$program"; then
+        PACKAGES_TO_INSTALL+=("$package")
+    fi
+}
+
+check_nodejs() {
+    echo
+    echo "Checking for node.js..."
+    echo -n "> "
+
+    local nvm_needed=true
+
+    if command_exists nodejs && command_exists npm; then
```

Comment:
This is command `node` on some systems, very angry that you installed a new version of node when I had a PERFECTLY FUNCTIONING install 😂.

Reply:
Note it’s a local installation that’s gone completely when you remove the dir.